test/cppsuite/test_harness/workload/database_operation.cxx

```diff
@@ -144,18 +145,21 @@ database_operation::insert_operation(thread_context *tc)
         uint64_t start_key = ccv[counter].coll.get_key_count();
         uint64_t added_count = 0;
         bool committed = true;
-        tc->transaction.begin(tc->session.get(), "");
+        tc->transaction.begin();
         while (tc->transaction.active() && tc->running()) {
             /* Insert a key value pair. */
             if (!tc->insert(ccv[counter].cursor, ccv[counter].coll.id, start_key + added_count)) {
                 committed = false;
                 break;
             }
             added_count++;
-            tc->transaction.try_commit(tc->session.get(), "");
+            tc->transaction.try_commit();
             /* Sleep the duration defined by the op_rate. */
             tc->sleep();
         }
+        /* Reset our cursor to avoid pinning content. */
+        testutil_check(ccv[counter].cursor->reset(ccv[counter].cursor.get()));
```

Comment:
I think this is doing two lookups. In general, we should try to avoid this with code like:
```
auto& c = ccv[counter].cursor;
testutil_check(c->reset(c.get()));
```

Reply:
Yeah true.